### DISCUSSION TITLE
Are EventWriters only called on master processes?

### DISCUSSION THREAD

**Discussion by AyushExel** [facebookresearch/detectron2#3604] [Q&A]
I want to perform certain operations that should not be repeated across other DDP processes( non 0 rank processes).  Can I use EventsWriters to create such functionalities and log the result to event storage for other processes to access? Or are there other recommend ways of doing this.

**Comment by ppwwyyxx** [comment]
EventWriters only run on master processes and https://detectron2.readthedocs.io/en/latest/modules/utils.html#detectron2.utils.comm.is_main_process is used to check.

**Reply by AyushExel** [reply]
Awesome. Thanks
